src/python/gudhi/sklearn/rips_persistence.py

```diff
@@ -0,0 +1,148 @@
+from sklearn.base import BaseEstimator, TransformerMixin
+
+import numpy as np
+# joblib is required by scikit-learn
+from joblib import Parallel, delayed
+
+# Mermaid sequence diagram - https://mermaid-js.github.io/mermaid-live-editor/
+# sequenceDiagram
+#     USER->>RipsPersistence: fit_transform(X)
+#     Note right of RipsPersistence: homology_dimensions=[i,j]
+#     RipsPersistence->>thread1: _tranform(X[0])
+#     RipsPersistence->>thread2: _tranform(X[1])
+#     Note right of RipsPersistence: ...
+#     thread1->>RipsPersistence: [array( Hi(X[0]) ), array( Hj(X[0]) )]
+#     thread2->>RipsPersistence: [array( Hi(X[1]) ), array( Hj(X[1]) )]
+#     Note right of RipsPersistence: ...
+#     RipsPersistence->>USER: [[array( Hi(X[0]) ), array( Hj(X[0]) )],<br/> [array( Hi(X[1]) ), array( Hj(X[1]) )],<br/> ...]
+
+
+class RipsPersistence(BaseEstimator, TransformerMixin):
+    """
+    This is a class for computing the persistence diagrams from a Vietoris-Rips complex.
+    """
+
+    def __init__(
+        self,
+        homology_dimensions,
+        max_edge_length=float('inf'),
+        input_type='point cloud',
+        nb_collapse=-1,
+        homology_coeff_field=11,
+        min_persistence=0.0,
+        expand_extra_dimension = True,
+        n_jobs=None,
+    ):
+        """
+        Constructor for the RipsPersistence class.
+
+        Parameters:
+            homology_dimensions (int or list of int): The returned persistence diagrams dimension(s).
+                Short circuit the use of :class:`~gudhi.representations.preprocessing.DimensionSelector` when only one
+                dimension matters (in other words, when `homology_dimensions` is an int).
+            max_edge_length (float): Rips value. Default is +Inf.
+            input_type (str): Can be 'point cloud' when inputs are point clouds, or 'lower distance matrix', when
+                inputs are lower triangular distance matrix (can be full square, but the upper part of the distance
+                matrix will not be considered). Default is 'point cloud'.
+            nb_collapse (int): The number of :func:`~gudhi.SimplexTree.collapse_edges` iterations to perform on the
+                SimplexTree. Default is -1, which means "automatic" (a relatively good enough number of iterations is
+                choosen).
+            homology_coeff_field (int): The homology coefficient field. Must be a prime number. Default value is 11.
+            min_persistence (float): The minimum persistence value to take into account (strictly greater than
+                `min_persistence`). Default value is `0.0`. Set `min_persistence` to `-1.0` to see all values.
+            expand_extra_dimension (bool): :func:`~gudhi.SimplexTree.expansion` is performed at
+                `max(homology_dimensions) + 1` if true, and `max(homology_dimensions)` otherwise. Default is true.
+            n_jobs (int): cf. https://joblib.readthedocs.io/en/latest/generated/joblib.Parallel.html
```

Comment:
We could maybe add a comment before the cf, such as (copied from scikitlearn's doc) : 

Number of jobs to run in parallel. None means 1 unless in a joblib.parallel_backend context. -1 means using all processors

Reply:
Ok, I added it on [bc99315](https://github.com/GUDHI/gudhi-devel/pull/691/commits/bc993151ba1fcd15067fe02c09734031ad1ebbcd)